pkg/utils/secrets/certificate_bundle.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"github.com/gardener/gardener/pkg/utils/infodata"
+)
+
+// DataKeyCertificateBundle is the key in the data map for the certificate bundle.
+const DataKeyCertificateBundle = "bundle.crt"
+
+// CertificateBundleSecretConfig is configuration for certificate bundles.
+type CertificateBundleSecretConfig struct {
+	Name            string
+	CertificatePEMs [][]byte
+}
+
+// CertificateBundle contains the name and the generated certificate bundle.
+type CertificateBundle struct {
+	Name   string
+	Bundle []byte
+}
+
+// GetName returns the name of the secret.
+func (s *CertificateBundleSecretConfig) GetName() string {
+	return s.Name
+}
+
+// Generate implements ConfigInterface.
+func (s *CertificateBundleSecretConfig) Generate() (DataInterface, error) {
+	return &CertificateBundle{
```

Comment:
Not sure, we don't have similar failures for other implementations of the interface, so for symmetry reasons I would keep it as is to make assumptions here of how the user would construct the `CertificateBundleSecretConfig`.

Reply:
Fine with me both. You can decide.
My reasoning here was: Fail early. If there is some client who messes up his construction of bundles in some cases because of lets say a race condition; we would get an error message instead of an empty bundle leading to runtime problems which are probably harder to debug.

Reply:
I would keep it as is with the reasoning stated above: Symmetry compared to other implementations of the interface, and not making any assumptions of how clients use this. If you think this should be generally improved (for all implementations) then please open a dedicated issue.